sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,479 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements a compositor to utilize forecasters based on ADI/CV categorization."""
+
+__author__ = ["shlok191"]
+
+import copy
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.croston import Croston
+from sktime.forecasting.naive import NaiveForecaster
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.adi_cv import ADICVTransformer
+
+adi_cv_transformer = ADICVTransformer(features=["class"])
+
+
+class CategoryCompositor(BaseForecaster):
+    """Compositor that utilizes varying forecasters by time series data's nature.
+
+    Applies a series-to-primitives transformer on a given time series. Based on the
+    generated value from the transformer, one of multiple forecasters provided by
+    the user in the form of a dictionary (key => category, value => forecaster) is
+    selected. Finally, the chosen forecaster is fit to the data for future predictions.
+
+    Parameters
+    ----------
+    forecasters : dict[sktime forecasters]
+        dict of forecasters with the key corresponding to categories generated
+        by the given transformer and the value corresponding to a sktime forecaster.
+
+    transformer : sktime transformer, default = ADICVTransformer()
+        A series-to-primitives sk-time transformer that generates a value
+        which can be used to quantify a choice of forecaster for the time series.
+
+    fallback_forecaster : sktime forecaster | None, Optional
+        A fallback forecaster that will be used if the category generated by
+        the transformer does not match any of the given forecasters.
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) are not given.
+
+    Examples
+    --------
+    >>> from sktime.forecasting.compose._category_compositor import CategoryCompositor
+    >>> from sktime.forecasting.croston import Croston
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.adi_cv import ADICVTransformer
+    >>> from sktime.transformations.series.tests.test_adi_cv import (
+    ...     _generate_erratic_series)
+    >>> category_compositor = CategoryCompositor(
```

Comment:
nice example! would you like to add comments or explanation what this does? it is not obvious for the user, and a good example is accompanied by a good explanation (comments or text)

Reply:
Ah, absolutely! I'll explain it in more detail. Sorry if this seems rushed, I've just been reading the LagLlama paper and have been very excited about getting started on adding that implementation as soon as I could... 😅